shield.go

```diff
@@ -9,27 +9,41 @@ import (
 	"golang.org/x/crypto/pbkdf2"
 )
 
-// Iterations is the number of iterations to run PBKDF2.
-const Iterations = 5000
-
-// MaxLength indicates the maximum length that a password can have.
-const MaxLength = 4096
+var ErrPasswordTooLong = errors.New("password too long")
 
-// SaltLength is the length of the salt generated by GenerateSalt.
-const SaltLength = 32
+// Shield is a struct that contains basic configuration variables used to
+// encrypt and check strings.
+//     iterations: is the number of iterations to run PBKDF2.
+//     maxLength: indicates the maximum length that a password can have.
+//     saltLength: is the length of the salt generated by GenerateSalt.
+type Shield struct {
+	iterations int
```

Comment:
Naming fields like this result in non-exported fields, thus the user can never explicitly set those values, it only works by passing the values in a positional way. Also the following is shorter :wink: 

```go
type Shield struct {
    Iterations, MaxLength, SaltLength int
}
```